provider/defaultingress/ingress.go

```diff
@@ -0,0 +1,453 @@
+		ingress, err = r.populateDefaultIngressFromList(ctx, state)
+		if err != nil {
+			return err
+		}
+	} else {
+		ingressResp, err := r.collection.Cluster(state.Cluster.ValueString()).Ingresses().Ingress(state.Id.ValueString()).Get().SendContext(ctx)
+		if err != nil {
+			return err
+		}
+		ingress = ingressResp.Body()
+	}
+
+	return r.populateState(ingress, state)
+}
+
+func (r *DefaultIngressResource) populateDefaultIngressFromList(ctx context.Context, state *DefaultIngress) (*cmv1.Ingress, error) {
+	ingresses, err := r.collection.Cluster(state.Cluster.ValueString()).Ingresses().List().SendContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	for _, ingress := range ingresses.Items().Slice() {
+		if ingress.Default() {
+			return ingress, nil
+		}
+	}
+
+	return nil, fmt.Errorf("failed to find default ingress")
+}
+
+func (r *DefaultIngressResource) populateState(ingress *cmv1.Ingress, state *DefaultIngress) error {
+	if state == nil {
+		state = &DefaultIngress{}
+	}
+	state.Id = types.StringValue(ingress.ID())
+	routeSelectors, ok := ingress.GetRouteSelectors()
+	var err error
+	if ok {
+		state.RouteSelectors, err = common.ConvertStringMapToMapType(routeSelectors)
+		if err != nil {
+			return err
+		}
+	}
+
+	excludedNamespaces, ok := ingress.GetExcludedNamespaces()
+	if ok {
+		state.ExcludedNamespaces, err = common.StringArrayToList(excludedNamespaces)
+		if err != nil {
+			return err
+		}
+	}
+	wp, ok := ingress.GetRouteWildcardPolicy()
+	if ok {
+		state.WildcardPolicy = types.StringValue(string(wp))
+	} else {
+		state.WildcardPolicy = types.StringNull()
+	}
+	rnmop, _ := ingress.GetRouteNamespaceOwnershipPolicy()
+	state.NamespaceOwnershipPolicy = types.StringValue(string(rnmop))
+	if ok {
+		state.NamespaceOwnershipPolicy = types.StringValue(string(rnmop))
+	} else {
+		state.NamespaceOwnershipPolicy = types.StringNull()
+	}
+	hostname, ok := ingress.GetClusterRoutesHostname()
+	if ok {
+		state.ClusterRoutesHostname = types.StringValue(hostname)
+	} else {
+		state.ClusterRoutesHostname = types.StringNull()
+	}
+	tls, ok := ingress.GetClusterRoutesTlsSecretRef()
+	if ok {
+		state.ClusterRoutesTlsSecretRef = types.StringValue(tls)
+	} else {
+		state.ClusterRoutesTlsSecretRef = types.StringNull()
+	}
+	state.LoadBalancerType = types.StringValue(string(ingress.LoadBalancerType()))
+
+	return nil
+}
+
+func (r *DefaultIngressResource) updateIngress(ctx context.Context, state, plan *DefaultIngress,
+	clusterId string, clusterCollection *cmv1.ClustersClient) error {
+
+	if state == nil {
+		state = &DefaultIngress{Cluster: plan.Cluster}
+	}
+	// In case default ingress was not part of state till now and we want to set
+	// it we need to bring it first as we need to set specific id
+	if common.IsStringAttributeUnknownOrEmpty(state.Id) {
+		err := r.populateDefaultIngress(ctx, state)
+		if err != nil {
+			return err
+		}
+	}
+
+	if !reflect.DeepEqual(state, plan) {
+		err := validateDefaultIngress(ctx, plan)
```

Comment:
Is there any reason why not adding this validation in the plan stage?

Reply:
not sure i understand, i assume it can be optimization but currently not sure how

Reply:
All validations that can be done in the plan mode (before calling the backend) need to be done than
I don't think this critical for now, lets open ticket to change it later